src/views/ifsBrowser.ts

```diff
@@ -153,11 +157,16 @@ class IFSDirectoryItem extends IFSItem {
 
   async getChildren(): Promise<BrowserItem[]> {
     const content = instance.getContent();
+    const showHidden = instance.getConfig()?.showHiddenFiles;
     if (content) {
       try {
         const objects = await content.getFileList(this.path, this.sort, handleFileListErrors);
-        const directories = objects.filter(o => o.type === `directory`);
-        const streamFiles = objects.filter(o => o.type === `streamfile`);
+        const directories = objects
+          .filter(o => o.type === `directory`)
+          .filter(o => showHidden || !o.name.startsWith(`.`) || alwaysShow(o.name));
+        const streamFiles = objects
+          .filter(o => o.type === `streamfile`)
+          .filter(o => showHidden || !o.name.startsWith(`.`) || alwaysShow(o.name));
```

Comment:
How about factorizing the filtering here?
```suggestion
    if (content) {
      try {
        const showHidden = instance.getConfig()?.showHiddenFiles;
        const filterIFSFile = (file:IFSFile, type: "directory" | "streamfile") => file.type === type && (showHidden || !file.name.startsWith(`.`) || alwaysShow(file.name));
        const objects = await content.getFileList(this.path, this.sort, handleFileListErrors);
        const directories = objects.filter(f => filterIFSFile(f, "directory"));
        const streamFiles = objects.filter(f => filterIFSFile(f, "streamfile"));
```

Reply:
@sebjulliand Did your parents nick-name you "The Typescript Zip Master"?! 😆 

Suggestion committed... 👍